### ISSUE TITLE
Generating strongly typed resource files not currently supported on .NET Core MSBuild 

### ISSUE THREAD

**Issue by livarcocc** [microsoft/msbuild#2272]
_From @iSazonov on July 6, 2017 13:41_

What is a right repo to ask about ResGen in `dotnet build`?

Currently I try to generate strongly typed resources and get error:
```C:\Users\user\AppData\Local\Microsoft\dotnet\sdk\2.0.0-preview2-006502\Microsoft.Common.CurrentVersion.targets(2921,5): error : Generating strongly typed resource files not currently supported on .NET Core MSBuild [C:\Users\user\Documents\GitHub\iSazonov\PowerShell\src\Microsoft.PowerShell.CoreCLR.Eventing\Microsoft.PowerShell.CoreCLR.Eventing.csproj]```

From related Issue https://github.com/dotnet/cli/issues/819:
> dotnet-resgen  got replaced by the  Microsoft.Net.Sdk  implementation, so closing this.

Is strongly typed resources supported?

_Copied from original issue: dotnet/cli#7084_

**Comment by livarcocc** [comment]
_From @wli3 on July 6, 2017 17:9_

Hi iSazonov,
Could you try to add resx files like this? https://github.com/dotnet/sdk/blob/cd25d6bbfc2aac61cc790c511900427ac64fdaa8/TestAssets/TestProjects/KitchenSink/TestApp/TestApp.csproj#L24
(This is a part of SDK automatic test asset)

**Comment by livarcocc** [comment]
_From @iSazonov on July 6, 2017 17:29_

@wli3 Thanks! Please clarify - I don't see strongly typed config in the csproj.

**Comment by livarcocc** [comment]
_From @wli3 on July 6, 2017 17:35_

Hi @iSazonov ,
I think I misunderstood you. Looks like your project is in Github, could you share the specific strongly typed config code link?

**Comment by livarcocc** [comment]
_From @iSazonov on July 6, 2017 17:41_

@wli3 I'll publish the sample tomorrow from my local branch.

**Comment by livarcocc** [comment]
_From @iSazonov on July 7, 2017 5:34_

I published my test. See last ``<ItemGroup``> in  https://github.com/iSazonov/PowerShell/blob/cproj-psversion-from-git-resx/src/Microsoft.PowerShell.CoreCLR.Eventing/Microsoft.PowerShell.CoreCLR.Eventing.csproj

Build command:
```
dotnet msbuild /t:"ResGen" /v:diag /noconsolelogger /fileLogger .\Microsoft.PowerShell.CoreCLR.Eventing.csproj
or
dotnet build /t:"ResGen" /v:diag /noconsolelogger /fileLogger .\Microsoft.PowerShell.CoreCLR.Eventing.csproj
```

I believe error place is [here](https://github.com/Microsoft/msbuild/blob/master/src/Tasks/GenerateResource.cs#L3262)
Is the project compiled without FEATURE_CODEDOM?

**Comment by livarcocc** [comment]
I am moving this issue over to MSBuild. FYI @rainersigwald

**Comment by rainersigwald** [comment]
Huh, we should have had an issue open for this already but I couldn't find one.

> Is the project compiled without FEATURE_CODEDOM?

Yes, it is: CodeDOM wasn't available at the time of the initial port to .NET Core and we haven't lit up this feature after the initial port.

It looks like CodeDOM is available in .NET Core 2.0 (for code-writing scenarios, though not compilation), so we should be able to enable that feature flag everywhere after we move to targeting .NET Core 2.0.

## Workarounds
Build with full framework MSBuild.exe (on a Windows machine).

Check in the (generated) `.cs` files and update them when necessary (from Windows/full framework).

**Comment by iSazonov** [comment]
@rainersigwald Thanks! 
[PowerShell Core](https://github.com/PowerShell/PowerShell/) is multiplatform project and we cannot use MSBuild.exe. Currently we use a [simple tool ](https://github.com/PowerShell/PowerShell/tree/master/src/ResGen)developed by @vors.

>It looks like CodeDOM is available in .NET Core 2.0 

Yes, it is. So I believe it is easy to fix the Issue.

**Comment by foo-baar** [comment]
Hello Team, I am sorry if I am asking silly question: So what I understand is that if I am using dotnet sdk on my Mac & vscode for editing, I can't create a strongly typed controller from db ?

(I know there is visual studio version now available for Mac).
Thanks 👍

**Comment by iSazonov** [comment]
Team, .Net Core 2.0 RTM was out - could you please milestone the Issue?

**Comment by kasper3** [comment]
Team, .Net Core 2.0 RTM was out - could you please milestone the Issue?

**Comment by hrushib** [comment]
@livarcocc, when is this feature going to be supported. Absence of this makes it difficult to build code from Linux, i.e. requires manual overhead of generating and checking in strongly typed resource access wrapper.

**Comment by rainersigwald** [comment]
From a quick look this morning, CodeDOM is **not** enough to bring this back: the task also depends on `System.Resources.ResXResourceReader` which is full-framework only.

**Comment by kasper3** [comment]
Should we open an API proposal for `ResXResourceReader` in CoreFX repo? https://github.com/dotnet/corefx/issues/12565 and https://github.com/Microsoft/msbuild/issues/1352 seem related issues.

**Comment by danmosemsft** [comment]
Discussed with @rainersigwald offline. I think it is worth trying to bring this back without waiting for ResXResourceReader (whose port is not scheduled currently)

My suspicion (just from browsing the code) is that ResxResourcrReader/ResXDataNode are not necessary for this scenario. StronglyTypedResourceBuilder hardly depends on them and it looks like those code paths can probably be removed. 

Happily MSBuild happens to already have a copy of [StronglyTypedResourceBuilder.cs](https://github.com/Microsoft/msbuild/blob/master/src/Tasks/StronglyTypedResourceBuilder.cs) because I guess historically they had a reason for avoiding a dependency on System.Design.dll. (Maybe it was only in the SDK?). I compared it with .NET Framework's current copy and it's basically the same. 

I think someone could take a shot at wiring this up now.

**Comment by danmosemsft** [comment]
@kasper3 any interst in taking a look?

**Comment by kasper3** [comment]
@danmosemsft, I tried building msbuild master on Windows 10, `msbuild\build.cmd -hostType Core` and getting errors like:

> C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.git_revwalk_next(RevWalkerHandle walker) [C:\Users\kasper\Source\Repos\msbuild\src\Samples\TaskUsageLogger\TaskUsageLogger.csproj]

Tried in three modes: Developer Command Prompt VS2017, normal cmd, normal PowerShell.

**Comment by danmosemsft** [comment]
Uh, I have no idea what that means. Is there any more information like callstack or message? Is this a fresh new clone - if not I would delete it and clone again?

**Comment by kasper3** [comment]
I just cloned the repo, tip of master branch. Call stack:

```
C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [C:\Users\kasper\Source\Repos\msbuild\src\Samples\XmlFileLogger\XmlFileLogger.csproj]
C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018: LibGit2Sharp.NotFoundException: object not found - no match for id (7c1f7a8cd8f2b844046892036912db28a7ec4c37) [C:\Users\kasper\Source\Repos\msbuild\src\Samples\XmlFileLogger\XmlFileLogger.csproj]
C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.Ensure.HandleError(Int32 result) [C:\Users\kasper\Source\Repos\msbuild\src\Samples\XmlFileLogger\XmlFileLogger.csproj]
C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.git_revwalk_next(RevWalkerHandle walker) [C:\Users\kasper\Source\Repos\msbuild\src\Samples\XmlFileLogger\XmlFileLogger.csproj]
C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.CommitLog.CommitEnumerator.MoveNext() [C:\Users\kasper\Source\Repos\msbuild\src\Samples\XmlFileLogger\XmlFileLogger.csproj]
C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Boolean& found) [C:\Users\kasper\Source\Repos\msbuild\src\Samples\XmlFileLogger\XmlFileLogger.csproj]
C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle..ctor(String projectDirectory, Repository repo, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot) [C:\Users\kasper\Source\Repos\msbuild\src\Samples\XmlFileLogger\XmlFileLogger.csproj]
C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot) [C:\Users\kasper\Source\Repos\msbuild\src\Samples\XmlFileLogger\XmlFileLogger.csproj]
C:\Users\kasper\.nuget\packages\nerdbank.gitversioning\2.1.23\build\Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner() [C:\Users\kasper\Source\Repos\msbuild\src\Samples\XmlFileLogger\XmlFileLogger.csproj]
```

```
git log -1
commit e8b480c373eadd9daa975c6e96e4dbbabcd9a4fe
```

**Comment by danmosemsft** [comment]
Shallow clone?
https://github.com/libgit2/libgit2sharp/issues/1351

**Comment by kasper3** [comment]
Yes, that was it.. I had `git clone <url> --single-branch --depth 1`, unshallowed it using https://stackoverflow.com/a/17937889 🐕 
Maybe something like https://github.com/dotnet/buildtools/pull/1977/files can be used to have safe rev-parsing, with `git-describe` command changed to `git describe --always --abbrev=0 --match "NOT A TAG"`, to get hash, since Libgit2 does not support shallow clones.

**Comment by rainersigwald** [comment]
Since our version numbers are commit-height based, we wouldn't support shallow clones regardless of libgit2sharp support. The experience could surely be better though, sorry you ran into this @kasper3.

**Comment by rainersigwald** [comment]
@kasper3 if it's helpful, I pushed my WIP to https://github.com/rainersigwald/msbuild/tree/strongly-typed-resgen.

I also ran under a debugger to see which of these codepaths was hit in the sample project https://github.com/iSazonov/PowerShell/blob/cproj-psversion-from-git-resx/src/Microsoft.PowerShell.CoreCLR.Eventing/Microsoft.PowerShell.CoreCLR.Eventing.csproj:

https://github.com/Microsoft/msbuild/blob/e8b480c373eadd9daa975c6e96e4dbbabcd9a4fe/src/Tasks/system.design/stronglytypedresourcebuilder.cs#L122-L143

Despite the "this will be rare after Whidbey M3" (that's Visual Studio **2005**) comment, that's the path that was traversed. If that's all that's ever used it'd make the port _much_ easier.

**Comment by kasper3** [comment]
@rainersigwald, thanks for the WIP branch. Seems like `StronglyTypedResourceBuilder.Create` method ends up requiring `ResXDataNode`, which is not available on .NET Core today. Should we port some code from https://referencesource.microsoft.com/#System.Windows.Forms/winforms/Managed/System/Resources/ResXDataNode.cs into MSBuild repo (internal types), unless there is a better workaround? :)

**Comment by rainersigwald** [comment]
@kasper3 That's what I was investigating under the debugger: it _looks_ like we could just `#if` out that code for .net core, and not lose any functionality. It didn't appear to be hit in the simple strongly-typed-resource scenario I was trying.

**Comment by kasper3** [comment]
After applying this patch on top of @rainersigwald's branch:

```diff
diff --git a/src/Tasks/system.design/stronglytypedresourcebuilder.cs b/src/Tasks/system.design/stronglytypedresourcebuilder.cs
index 332ff638..49976ec2 100644
--- a/src/Tasks/system.design/stronglytypedresourcebuilder.cs
+++ b/src/Tasks/system.design/stronglytypedresourcebuilder.cs
@@ -119,27 +119,11 @@ namespace Microsoft.Build.Tasks
             Dictionary<String, ResourceData> resourceTypes = new Dictionary<String, ResourceData>(StringComparer.InvariantCultureIgnoreCase);
             foreach (DictionaryEntry de in resourceList)
             {
-                ResXDataNode node = de.Value as ResXDataNode;
-                ResourceData data;
-                if (node != null)
-                {
-                    string keyname = (string)de.Key;
-                    if (keyname != node.Name)
-                        throw new ArgumentException(SR.GetString(SR.MismatchedResourceName, keyname, node.Name));
-
-                    String typeName = node.GetValueTypeName((AssemblyName[])null);
-                    Type type = Type.GetType(typeName);
-                    String valueAsString = node.GetValue((AssemblyName[])null).ToString();
-                    data = new ResourceData(type, valueAsString);
-                }
-                else
-                {
-                    // If the object is null, we don't have a good way of guessing the
-                    // type.  Use Object.  This will be rare after WinForms gets away
-                    // from their resource pull model in Whidbey M3.
-                    Type type = (de.Value == null) ? typeof(Object) : de.Value.GetType();
-                    data = new ResourceData(type, de.Value == null ? null : de.Value.ToString());
-                }
+                // If the object is null, we don't have a good way of guessing the
+                // type.  Use Object.  This will be rare after WinForms gets away
+                // from their resource pull model in Whidbey M3.
+                Type type = (de.Value == null) ? typeof(Object) : de.Value.GetType();
+                var data = new ResourceData(type, de.Value == null ? null : de.Value.ToString());
                 resourceTypes.Add((String)de.Key, data);
             }
 
@@ -260,39 +244,6 @@ namespace Microsoft.Build.Tasks
             return ccu;
         }
 
-        internal static CodeCompileUnit Create(String resxFile, String baseName, String generatedCodeNamespace, CodeDomProvider codeProvider, bool internalClass, out String[] unmatchable)
-        {
-            return Create(resxFile, baseName, generatedCodeNamespace, null, codeProvider, internalClass, out unmatchable);
-        }
-
-        [SuppressMessage("Microsoft.Naming", "CA1704:IdentifiersShouldBeSpelledCorrectly")]
-        internal static CodeCompileUnit Create(String resxFile, String baseName, String generatedCodeNamespace, String resourcesNamespace, CodeDomProvider codeProvider, bool internalClass, out String[] unmatchable)
-        {
-            if (resxFile == null)
-                throw new ArgumentNullException("resxFile");
-
-            // Read the resources from a ResX file into a dictionary - name & type name
-            Dictionary<String, ResourceData> resourceList = new Dictionary<String, ResourceData>(StringComparer.InvariantCultureIgnoreCase);
-            using (ResXResourceReader rr = new ResXResourceReader(resxFile))
-            {
-                rr.UseResXDataNodes = true;
-                foreach (DictionaryEntry de in rr)
-                {
-                    ResXDataNode node = (ResXDataNode)de.Value;
-                    String typeName = node.GetValueTypeName((AssemblyName[])null);
-                    Type type = Type.GetType(typeName);
-                    String valueAsString = node.GetValue((AssemblyName[])null).ToString();
-                    ResourceData data = new ResourceData(type, valueAsString);
-                    resourceList.Add((String)de.Key, data);
-                }
-            }
-
-            // Note we still need to verify the resource names are valid language
-            // keywords, etc.  So there's no point to duplicating the code above.
-
-            return InternalCreate(resourceList, baseName, generatedCodeNamespace, resourcesNamespace, codeProvider, internalClass, out unmatchable);
-        }
-
         private static void AddGeneratedCodeAttributeforMember(CodeTypeMember typeMember)
         {
             CodeAttributeDeclaration generatedCodeAttrib = new CodeAttributeDeclaration(new CodeTypeReference(typeof(System.CodeDom.Compiler.GeneratedCodeAttribute)));
```

I am getting `SR is inaccessible due to its protection level`.

**Comment by danmosemsft** [comment]
I can't see the rest is the code on my phone but my guess is that the STR class is pulling some resource strings  using SR.XXX and needs to change to follow however rest of MSBuild code does it?

**Comment by nathansoz** [comment]
Did this issue make any forward progress since the last post? I'm looking into this for a project I'm working on and wondering if we should expect a fix soon or if we should roll a workaround.

**Comment by danmosemsft** [comment]
@nathansoz I don't think so. I suspect they'd welcome a PR. It seemed at the time like there was likely not much work.

**Comment by dasMulli** [comment]
I assume this now impacts the winforms/wpf-on-.net-core scenario?
Would this require msbuild to run on the winforms/wpf shared framework to get the required generation/seralization assemblies used on the full-fx msbuild version? Or does it need a clean re-implementation to make sure the resulting assemblies are compatible for both net* and netcoreapp* applications built using Core MSBuild?

**Comment by rainersigwald** [comment]
> Just to clarify, does this mean we are still unable to generate `resx` file with `msbuild` on .NET Core at build time?

Correct. We expect to resolve this with .NET Core 3.0. Most of the discussion on that is going on in #2221.

**Comment by rainersigwald** [comment]
I think it's possible to do this part (generate C# from the resx) without the full task. #3819 is an attempt to do so.

In the end we should make sure both scenarios work, so I'd keep both open but I don't have strong feelings about it.

**Comment by bergmeister** [comment]
@rainersigwald I see you've already completed some work in #4420 but you also call out some work such as 'Check to ensure that the reader type is available when not targeting netcoreapp3.0' (which is ticked now).
Does this mean that a project targeting .Net Standard (2.0) can already auto-generate the `*.Designer.cs` files (in my case just the simple case of only strings)? Using `preview8,` it does not seem to that at the moment or is a property needed to opt into this at the moment for `netstandard2.0` projects? Thanks.

**Comment by rainersigwald** [comment]
@bergmeister That wasn't directly related to code generation, and StronglyTypedResourceGenerator doesn't work in preview8. #4588 should enable it in the future, though.

If your code is open source, I can add it to the set of things I'm testing before merging that PR. Just feed me a link :)

**Comment by bergmeister** [comment]
@rainersigwald Thanks for the quick answer. An example project that you could use for testing is this one: https://github.com/PowerShell/PSScriptAnalyzer/tree/master/Engine
Just remove the `Strings.Designer.cs` file, remove the `global.json` one directory above and then just type `dotnet build` (all NuGet packages are on NuGet.org so everything should just build out of the box).

**Comment by bergmeister** [comment]
@rainersigwald I tried to use `Preview9` on my project (links in last comment) but `dotnet build` does not create me the strongly typed .cs file from .resx file although the csproj has the following:
```
  <ItemGroup>
    <Compile Update="Strings.Designer.cs">
      <DesignTime>True</DesignTime>
      <AutoGen>True</AutoGen>
      <DependentUpon>Strings.resx</DependentUpon>
    </Compile>
  </ItemGroup>

  <ItemGroup>
    <EmbeddedResource Update="Strings.resx">
      <Generator>ResXFileCodeGenerator</Generator>
      <LastGenOutput>Strings.Designer.cs</LastGenOutput>
    </EmbeddedResource>
  </ItemGroup>
```
Am I am missing something, is there something that needs to be enabled explicitly for a .net standard (2.0) library?

**Comment by rainersigwald** [comment]
@bergmeister Yes, this is fairly confusing: your build was configured so that building _inside Visual Studio_ would regenerate the source files, but building _on the command line_ would not (regardless of whether it was `msbuild.exe` or `dotnet build` doing it).

When I tested your repo I applied this patch to change to build-time generation in all cases:

```diff
diff --git a/Engine/Engine.csproj b/Engine/Engine.csproj
index 560ce8d..2fca8db 100644
--- a/Engine/Engine.csproj
+++ b/Engine/Engine.csproj
@@ -46,18 +46,19 @@
     <PackageReference Include="Microsoft.CSharp" Version="4.5.0" />
   </ItemGroup>
 
-  <ItemGroup>
+  <!-- <ItemGroup>
     <Compile Update="Strings.Designer.cs">
       <DesignTime>True</DesignTime>
       <AutoGen>True</AutoGen>
       <DependentUpon>Strings.resx</DependentUpon>
     </Compile>
-  </ItemGroup>
+  </ItemGroup> -->
 
   <ItemGroup>
     <EmbeddedResource Update="Strings.resx">
       <Generator>ResXFileCodeGenerator</Generator>
-      <LastGenOutput>Strings.Designer.cs</LastGenOutput>
+      <StronglyTypedFileName>$(IntermediateOutputPath)\Strings.Designer.cs</StronglyTypedFileName>
+      <StronglyTypedLanguage>CSharp</StronglyTypedLanguage>
     </EmbeddedResource>
   </ItemGroup>
```

**Comment by bergmeister** [comment]
@rainersigwald Thanks for the info, however, even with the `rc1` of .Net Core 3 and your local changes, it does not work for me

**Comment by rainersigwald** [comment]
@bergmeister Can you please file a new issue with details? How exactly does it fail to work--build error? Runtime failure doing something in particular?

**Comment by bergmeister** [comment]
Ok, I can create an issue with a minimized repro.
It fails on build due to the `Strings.Designer.cs` not being created by the build (I deleted the file locally as I'd expect `dotnet build` to create it automatically, is my assumption here right?)

**Comment by rainersigwald** [comment]
Yes, that's correct--the copy in the checked-in tree is not needed when switching over to build-time generation.

**Comment by rainersigwald** [comment]
@bergmeister, I think I've reproduced the problem; it looks like something's gone wrong with the namespace/class name that we're generating. Don't worry about minimizing but please do file a bug.

**Comment by rainersigwald** [comment]
@bergmeister with this I can get your project to compile:

```diff
diff --git a/Engine/Engine.csproj b/Engine/Engine.csproj
index 560ce8d..e1f143b 100644
--- a/Engine/Engine.csproj
+++ b/Engine/Engine.csproj
@@ -46,18 +46,21 @@
     <PackageReference Include="Microsoft.CSharp" Version="4.5.0" />
   </ItemGroup>
 
-  <ItemGroup>
+  <!-- <ItemGroup>
     <Compile Update="Strings.Designer.cs">
       <DesignTime>True</DesignTime>
       <AutoGen>True</AutoGen>
       <DependentUpon>Strings.resx</DependentUpon>
     </Compile>
-  </ItemGroup>
+  </ItemGroup> -->
 
   <ItemGroup>
     <EmbeddedResource Update="Strings.resx">
       <Generator>ResXFileCodeGenerator</Generator>
-      <LastGenOutput>Strings.Designer.cs</LastGenOutput>
+      <StronglyTypedFileName>$(IntermediateOutputPath)\Strings.Designer.cs</StronglyTypedFileName>
+      <StronglyTypedLanguage>CSharp</StronglyTypedLanguage>
+      <StronglyTypedNamespace>Microsoft.Windows.PowerShell.ScriptAnalyzer</StronglyTypedNamespace>
+      <StronglyTypedClassName>Strings</StronglyTypedClassName>
     </EmbeddedResource>
   </ItemGroup>

 ```

We can chase down exactly why all this is necessary in your bug.

**Comment by bergmeister** [comment]
Thanks, it works now with your 2 new lines.
The `StronglyTypedNamespace` line seems to be only necessary because we have the following in the csproj (from the days when we converted from project.json to the new csproj format to make it work using Visual Studio)
`<RootNamespace>Microsoft.Windows.PowerShell.ScriptAnalyzer</RootNamespace>`
Now that the command line works, we don't care about Visual Studio too much tbh but I will check that later today (using VS 16.2.5 somehow the setting to allow VS to use preview SDKs has gone or can I use the VS Preview for that now only?)
Is the `StronglyTypedClassName` property needed due to the resx file not matching the project name? I tried renaming it to `Engine.resx` and changed the references in the csproj but the property was still needed then.
Should I file the 'bug' to be specific about this project or rather generic that those properties seem to be needed? As a user I can be flexible, but I'd like to know what is the cleanest/simplest csproj declaration to have this working (optionally also in a way that it works with VS)
